src/main/flight/pid.c

```diff
@@ -89,26 +89,32 @@ FAST_DATA_ZERO_INIT float throttleBoost;
 pt1Filter_t throttleLpf;
 #endif
 
-PG_REGISTER_WITH_RESET_TEMPLATE(pidConfig_t, pidConfig, PG_PID_CONFIG, 3);
+PG_REGISTER_WITH_RESET_FN(pidConfig_t, pidConfig, PG_PID_CONFIG, 4);
 
+void pgResetFn_pidConfig(pidConfig_t *pidConfig)
+{
+    unsigned denom = 1;
 #if defined(STM32F411xE)
```

Comment:
Only for f411? Also this should be a target.h define if so. USE_BMI270_DENOM_CHECK

Reply:
Will that work with all the multi defined gyro's in config.h. We need to know / check which one is activated.

Reply:
Maybe instead of BMI270, simply GYRO

Then provide an array of gyros that need this check...

Reply:
Works on BETAFPVF4SX1280 (BMI270) but not on CRAZYBEEF4SX1280 (ICM42688P).

- Looks like initialization on the ICM42688P is not ready yet.
- Looking at the values in status it's detected.

Reply:
Removed processor macros and added check for intended targets.
Moved logic to another place and now it works as intended.